arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorizedArrowReader.java

```diff
@@ -463,12 +460,16 @@ public static VectorizedArrowReader positionsWithSetArrowValidityVector() {
     return new PositionVectorReader(true);
   }
 
-  private static final class NullVectorReader extends VectorizedArrowReader {
-    private static final NullVectorReader INSTANCE = new NullVectorReader();
+  public static final class NullVectorReader extends VectorizedArrowReader {
+
+    public NullVectorReader(Types.NestedField icebergField) {
+      super(icebergField);
+    }
 
     @Override
     public VectorHolder read(VectorHolder reuse, int numValsToRead) {
-      return VectorHolder.dummyHolder(numValsToRead);
+      NullVector vector = new NullVector(icebergField().name(), numValsToRead);
+      return new VectorHolder.NullVectorHolder(vector, icebergField(), numValsToRead);
```

Comment:
there are places in the codebase that relied on `ConstantVectorHolder` so I'm not sure whether we should introduce a separate `NullVectorHolder`.
I tried it with the following diff, which should work too:
```
--- a/arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java
+++ b/arrow/src/main/java/org/apache/iceberg/arrow/vectorized/VectorHolder.java
@@ -19,6 +19,7 @@
 package org.apache.iceberg.arrow.vectorized;

 import org.apache.arrow.vector.FieldVector;
+import org.apache.arrow.vector.NullVector;
 import org.apache.iceberg.MetadataColumns;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.types.Type;
@@ -138,16 +139,19 @@ public class VectorHolder {
   public static class ConstantVectorHolder<T> extends VectorHolder {
     private final T constantValue;
     private final int numRows;
+    private final FieldVector vector;

     public ConstantVectorHolder(int numRows) {
       this.numRows = numRows;
       this.constantValue = null;
+      this.vector = new NullVector("_dummy_", numRows);
     }

     public ConstantVectorHolder(Types.NestedField icebergField, int numRows, T constantValue) {
       super(icebergField);
       this.numRows = numRows;
       this.constantValue = constantValue;
+      this.vector = new NullVector(icebergField.name(), numRows);
     }

     @Override
@@ -158,6 +162,11 @@ public class VectorHolder {
     public Object getConstant() {
       return constantValue;
     }
+
+    @Override
+    public FieldVector vector() {
+      return vector;
+    }
```